bin/ci/java-feign.json

```diff
@@ -3,7 +3,7 @@
   "generatorName": "java",
   "inputSpec": "modules/openapi-generator/src/test/resources/2_0/petstore-with-fake-endpoints-models-for-testing.yaml",
   "outputDir": "samples/client/petstore/java/feign",
-  "templateDir": "modules/openapi-generator/src/main/resources/Java/libraries/feign",
```

Comment:
Note these removals of `libraries/<library name>` in the Java sample configs and scripts.

While this syntax works if the entirety of the library code is in the named folder, this would cause fallbacks to fail if anyone were to remove something like `model.mustache` intending to fallback to `resources/Java/model.mustache`. We should avoid this possible confusion especially when these subdirectories are structured to follow our library lookup logic:

* User customized library path (e.g. `custom_template/libraries/feign/model.mustache`)
* User customized generator top-level path (e.g. `custom_template/model.mustache`)
* Embedded library path (e.g. `resources/Java/libraries/feign/model.mustache`)
* Embedded top-level path (e.g. `resources/Java/libraries/feign/model.mustache`)
* Common embedded path (e.g. `resources/_common/model.mustache`)
* throw `TemplateNotFoundException`